cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -365,3 +365,98 @@ Given('device is rebooted', () => {
     throw new Error(CONSTANTS.STEP_IMPLEMENTATION_MISSING);
   });
 });
+
+/**
+ * @module fireboltCalls
+ * @function And 3rd party '(.+)' app is dismissed
+ * @description To dismiss the launched app
+ * @param {String} app - app name.
+ * @example
+ * And 3rd party 'firebolt' app is dismissed
+ */
+Given(/3rd party '(.+)' app is dismissed$/, async (appType) => {
+  const appId = Cypress.env(CONSTANTS.RUNTIME).appId;
+  let KeyPressSequence;
+  if (
+    Cypress.env(CONSTANTS.RUNTIME) &&
+    Cypress.env(CONSTANTS.RUNTIME).intent &&
+    Cypress.env(CONSTANTS.RUNTIME).intent.keyPressSequence
+  ) {
+    KeyPressSequence = Cypress.env(CONSTANTS.RUNTIME).intent.keyPressSequence;
+  } else if (
+    Cypress.env('app_metadata') &&
+    Cypress.env('app_metadata')[appId] &&
+    Cypress.env('app_metadata')[appId].defaultKeyPressSequence
+  ) {
+    KeyPressSequence = Cypress.env('app_metadata')[appId].defaultKeyPressSequence;
+  } else if (Cypress.env('app_metadata') && Cypress.env('app_metadata').defaultKeyPressSequence) {
+    KeyPressSequence = Cypress.env('app_metadata').defaultKeyPressSequence;
+  } else {
+    throw new Error(`Expected KeyPressSequence was not found for ${appId} in app_metadata.json`);
+  }
+  cy.exitAppSession('dismissApp', KeyPressSequence.dismiss).then((response) => {
+    fireLog.info(`Response from platform: ${JSON.stringify(response)}`);
+  });
+});
+
+/**
+ * @module fireboltCalls
+ * @function And 3rd party '(.+)' app should be exited
+ * @description To validate that the app is dismissed
+ * @param {String} app - app name.
+ * @example
+ * Then 3rd party 'firebolt' app should be exited
+ */
+Given(/3rd party '(.+)' app should be exited$/, async (app) => {
+  // getAppState validation
+  const appId = Cypress.env(CONSTANTS.CURRENT_APP_ID);
+  const requestMapForGetAppState = {
+    method: CONSTANTS.REQUEST_OVERRIDE_CALLS.GETAPPSTATE,
+    params: appId,
+  };
+  fireLog.info(
+    `Sending request to fetch ${appId} app state: ${JSON.stringify(requestMapForGetAppState)}`
+  );
+  cy.sendMessagetoPlatforms(requestMapForGetAppState)
+    .then((response) => {
+      const responseString = JSON.stringify(response);
+      if (response && response.appState && response.visibilityState) {
+        if (
+          response.appState.toUpperCase() === CONSTANTS.LIFECYCLE_STATES.INACTIVE &&
+          response.visibilityState.toUpperCase() === CONSTANTS.VISIBLE
+        ) {
+          fireLog.info(
+            `State validation successful: Current state of ${appId} app is ${responseString} as expected`
+          );
+        } else if (
+          !response.appState &&
+          response.visibilityState.toUpperCase() === CONSTANTS.VISIBLE
+        ) {
+          cy.log(
+            `State validation successful: Current state of ${appId} app is ${responseString} as expected`
+          );
+        } else {
+          fireLog.fail(`${appId} app is not dismissed. Response :${responseString}`);
+        }
+      }
+    })
```

Comment:
else condition for line 423 is missing

Reply:
removed the main if condition and added sub if-else